sdk/communication/azure-communication-chat/azure/communication/chat/_communication_identifier_serializer.py

```diff
@@ -3,130 +3,102 @@
-            if not identifierModel.phone_number:
-                raise ValueError("PhoneNumberIdentifier must have a valid attribute - phone_number")
-            return PhoneNumberIdentifier(identifierModel.phone_number.value, raw_id=raw_id)
-        if identifierModel.microsoft_teams_user is not None:
-            if identifierModel.microsoft_teams_user.is_anonymous not in [True, False]:
-                raise ValueError("MicrosoftTeamsUser must have a valid attribute - is_anonymous")
-            if not identifierModel.microsoft_teams_user.user_id:
-                raise ValueError("MicrosoftTeamsUser must have a valid attribute - user_id")
-            if not identifierModel.microsoft_teams_user.cloud:
-                raise ValueError("MicrosoftTeamsUser must have a valid attribute - cloud")
-            return MicrosoftTeamsUserIdentifier(
-                raw_id=raw_id,
-                user_id=identifierModel.microsoft_teams_user.user_id,
-                is_anonymous=identifierModel.microsoft_teams_user.is_anonymous,
-                cloud=identifierModel.microsoft_teams_user.cloud
-            )
-
-        return UnknownIdentifier(raw_id)
-
-    @classmethod
-    def _getIdentifierType(cls, communicationIdentifier):
-        def has_attributes(obj, attributes):
-            return all([hasattr(obj, attr) for attr in attributes])
-
-        if has_attributes(communicationIdentifier, ["identifier"]):
-            return _IdentifierType.COMMUNICATION_USER_IDENTIFIER
-
-        if has_attributes(communicationIdentifier, ['phone_number', 'raw_id']):
-            return _IdentifierType.PHONE_NUMBER_IDENTIFIER
-
-        if has_attributes(communicationIdentifier, ["raw_id", "user_id", "is_anonymous", "cloud"]):
-            return _IdentifierType.MICROSOFT_TEAMS_IDENTIFIER
-
-        if has_attributes(communicationIdentifier, ["raw_id"]):
-            return _IdentifierType.UNKNOWN_IDENTIFIER
+        return CommunicationIdentifierModel(
+            raw_id=identifier.raw_id
+        )
+    except AttributeError:
+        raise TypeError("Unsupported identifier type " + identifier.__class__.__name__)
+
+
+def deserialize_identifier(identifier_model):
+    # type: (CommunicationIdentifierModel) -> CommunicationIdentifier
+    """
+    Deserialize the CommunicationIdentifierModel into Communication Identifier
+
+    :param identifier_model: CommunicationIdentifierModel
+    :type identifier_model: CommunicationIdentifierModel
+    :return: CommunicationIdentifier
+    """
+    raw_id = identifier_model.raw_id
+    if not raw_id:
+        raise ValueError("Identifier must have a valid ID")
+
+    _assert_maximum_one_nested_model(identifier_model)
+
+    if identifier_model.communication_user:
+        return CommunicationUserIdentifier(raw_id, raw_id=raw_id)
+    if identifier_model.phone_number:
+        if not identifier_model.phone_number:
+            raise ValueError("PhoneNumberIdentifier must have a valid attribute - phone_number")
```

Comment:
This is only called as part of deserialization of responses, correct? If so, we do not want to validate service responses. Service responses.

Reply:
Yeah I was thinking this while I was refactoring the code.... maybe I will take it out while I'm removing things :)

Reply:
We have this validation in the other languages. Where would you want the code to throw if we're not validating the service response here? I get the sentiment, but I like that our SDKs actually help validating the service in live tests. This is how we were able to discover multiple bugs during development.

@RezaJooyandeh